### ISSUE TITLE
Spy can get trapped between Ninjas, grid boundaries, or Rooms

### ISSUE THREAD

**Issue by jtara1** [Jester565/SpyNinjaGame#19]
I think Alex mentioned this in (library) meeting number 2.

Right now the UI will recursively ask and attempt to move the Spy when the Spy's MoveStatus from attempting to move is ILLEGAL which can happen when attempting to move on the following:

* out of bounds
* any side of the room except the top
* ninja

so it's possible (probably very unlikely) the Spy gets trapped between any of the above listed, the recursion never ends, and the game state is just stuck forever.

e.g.:

```
[ ][ ][ ][ ][N][ ][ ][ ][ ]     Lives: 3
[ ][R][ ][ ][R][ ][ ][R][ ]     Ammo: 0
[ ][N][ ][I][ ][ ][ ][ ][ ]     Invincible Turns: 0
[N][ ][ ][ ][ ][ ][ ][ ][ ]     Radar: Disabled
[S][R][r][ ][R][ ][ ][b][ ]
[N][ ][ ][ ][N][B][ ][ ][ ]
[ ][ ][ ][ ][ ][ ][ ][ ][ ]
[ ][R][ ][ ][R][ ][ ][R][ ]
[ ][ ][ ][ ][N][ ][ ][ ][ ]
```

**Comment by frfabianrp** [comment]
Shouldn't the Spy lose when moving into the space occupied by a Ninja? That's how the game handled that move before

**Comment by Jester565** [comment]
Oh dang!  That is one good test case!  I have an idea...  Do you want to
take this issue because I'd be happy to handle it?

**Comment by Jester565** [comment]
Francisco see issue #15 and issue #17

**Comment by jtara1** [comment]
@frfabianrp 

tl;dr professor said it's handled this way 

---

@Jester565 

you want to be assigned this issue? go for it

**Comment by Jester565** [comment]
I fixed it in this commit: d0065f23a0f3c5c7cf1a359fef12afbf78b4be35 .  However, I have not tested it yet.
